### PR TITLE
use HashMap::entry

### PR DESCRIPTION
**Pull request by techno-tanoC**
use [`HashMap::entry`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry)

### CODE REVIEW

File: src/handler/websocket.rs [hunk 1]
```diff
@@ -80,11 +80,5 @@ async fn setup_sender(state: &super::super::AppState, group_id: &str) -> broadca
...
     let group_id = group_id.to_string();
 
-    match group_list.get_mut(&group_id) {
-        None => {
-            group_list.insert(group_id, tx.clone());
-            tx
-        }
-        Some(group) => group.clone(),
-    }
+    group_list.entry(group_id).or_insert(tx).clone()
```
Comment (techno-tanoC) [inline src/handler/websocket.rs:83]: Do you prefer [`or_insert_with`](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with) ?

Comment (Taillook) [reply src/handler/websocket.rs:83]:
```
    group_list.entry(group_id).or_insert_with(|| {
        let (tx, _rx) = broadcast::channel(100);
        tx
    }).clone()
```

I thought that writing it this way would be a saving memory in temporary.

Comment (techno-tanoC) [reply src/handler/websocket.rs:83]: Good! I use it.

File: src/handler/websocket.rs [hunk 2]
```diff
@@ -76,15 +76,11 @@ async fn websocket(stream: WebSocket, state: Arc<super::super::AppState>, params
...
-        None => {
-            group_list.insert(group_id, tx.clone());
-            tx
-        }
-        Some(group) => group.clone(),
-    }
+    group_list.entry(group_id).or_insert_with(|| {
+        let (tx, _rx) = broadcast::channel(100);
+        tx
+    }).clone()
```
Comment (Taillook) [inline src/handler/websocket.rs:85]: to tuple subscript access.

```suggestion
    group_list
        .entry(group_id)
        .or_insert_with(|| broadcast::channel(100).0)
        .clone()
```

**Review by Taillook — APPROVED** [review]

**Pull request merged**